### PR TITLE
MGMT-4509 Format memory units according to capacity in memory validation messages

### PR DESCRIPTION
**Pull request by jordigilh**
This PR adds 2 changes:
* Formats the message displayed in the host memory validation to utilize the closest capacity type available based on the amount of memory of the host. For instance: if the host has 100 MiB of memory, it will show as such, rather than a fixed 0 GiB as it is right now. Example:
`Require at least 8 GiB RAM role worker, found only 0`
With this PR, the message becomes:
`Require at least 8.35 GiB RAM for role worker, found only 100 MiB`
* Adds two decimals for GiB, TiB and PiB so that the message uttered by the memory validation contains more detailed information on why the validation failed. Without decimals, it can end up showing a message like this one:
 `Require at least 8 GiB RAM role worker, found only 8 GiB` 
instead of 
`Require at least 8.35 GiB RAM for role worker, found only 8.00 GiB` 
as shown with the changes in this PR.

Also, this change aligns with the UI display for memory units which includes 2 decimal digits.

@jakub-dzon @machacekondra @masayag @pkliczewski @lalon4 please review at your discretion.

Thanks,

/Jordi

### CODE REVIEW

**Comment by avishayt** [comment]
This is a very nice addition, thanks!
Maybe this should go to conversions for more general use?
https://github.com/openshift/assisted-service/pull/1297

**Comment by pkliczewski** [comment]
Looks good!

**Comment by jordigilh** [comment]
> This is a very nice addition, thanks!
> Maybe this should go to conversions for more general use?
> #1297

Yeah, makes sense. I will update the PR. :smile:

File: pkg/conversions/conversions_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestConversions(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Conversions tests Suite")
+}
+
+var _ = Describe("bytes to string conversion", func() {
```
Comment (jakub-dzon) [inline pkg/conversions/conversions_test.go:16]: It would be great to have a few tests for uneven amounts (i.e. 1229 B, etc.).

Comment (jordigilh) [reply pkg/conversions/conversions_test.go:16]: Sure. I'll add a case :smile:

**Review by jakub-dzon — APPROVED** [review]

**Review by avishayt — APPROVED** [review]

**Review by pkliczewski — APPROVED** [review]

**Comment by jordigilh** [comment]
Ok, done with this one. I had last minute changes to rework on declaring the figures for the new tests. But it's done, really :smile:.

**Review by avishayt — APPROVED** [review]

**Pull request merged**